### PR TITLE
improve type error message

### PR DESCRIPTION
**Pull request by cherifGsoul**
For https://github.com/canjs/canjs/issues/5310

Improve the error message in `check` function.

Instead of:
`Uncaught Error: "3" is not of type Number.`

Now:
`Uncaught Error: 3 (String) is not of type Number.`

### CODE REVIEW

File: can-type.js [hunk 1]
```diff
@@ -45,7 +46,9 @@ function booleanNew(value) {
 var maybeValues = Object.freeze([null, undefined]);
 
 function check(Type, val) {
-	throw new Error('Type value ' + typeof val === "string" ? '"' + val + '"' : val + ' is not of type ' + canReflect.getName(Type) + '.'	);
+	var valueType = typeof val;
+	valueType = valueType.charAt(0).toUpperCase() + valueType.slice(1);
```
Comment (matthewp) [inline can-type.js]: can you do `canReflect.getName(Type[baseTypeSymbol])` to get the name? If so that might be better.

Comment (cherifGsoul) [reply can-type.js]: @matthewp `Type` is just a type constructor without symbols in this case, `canReflect.getName(Type[baseTypeSymbol])` returns `undefined`.

Comment (matthewp) [reply can-type.js]: ok, I guess this is fine then.

**Comment by matthewp** [comment]
Looks good!

**Pull request merged**